### PR TITLE
FTL on a spaced tile yeets instead of gibs

### PR DESCRIPTION
**Pull request by Varen**
## About the PR
Being on a space tile when FTL starts or ends currently gibs you. This changes it so you're instead thrownaway from the shuttle center, if you're not buckled.

## Why / Balance
Being on any shuttle with lattice on the interior deletes you from the game with no warning. A character on a shuttle that's been bombed getting gibbed, a salvager on the wrong tile, a number of situations that should be recoverable instead result in instant death. 
As the reason of this was originally done to avoid people riding shuttles on the outside, this still solves that issue without impacting the gameplay of other shuttle usage.

## Technical details
On FTL, if unbuckled and dropped down, check if the body is on a spaced tile, and if so, throw the body in a direction away from the shuttle.

## Media

https://github.com/space-wizards/space-station-14/assets/3650235/8efefcd2-cddb-4370-b8fd-d4c7737b5603

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
N/A

**Changelog**

:cl:
- tweak: FTL transit on damaged shuttles is not lethal anymore, but passengers should still hold on tight and remain inside the vehicle.

### CODE REVIEW

**Comment by deltanedas** [comment]
funny

File: Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs [hunk 1]
```diff
@@ -507,26 +502,42 @@ private void KnockOverKids(TransformComponent xform, ref ValueList<EntityUid> to
...
+            }
+
+            // Mass independant yeeting.
+            var impulseVector = direction.Normalized() * 500.0f * phys.Mass;
+            _physics.ApplyLinearImpulse(tossed, impulseVector, body: phys);
         }
+
+
+
     }
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs:541]: Use throwing code my guy, this is a lot slower. Also don't use GetValues.

File: Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs [hunk 2]
```diff
@@ -81,13 +81,15 @@ public sealed partial class ShuttleSystem
...
     private void InitializeFTL()
     {
         _bodyQuery = GetEntityQuery<BodyComponent>();
         _buckleQuery = GetEntityQuery<BuckleComponent>();
         _statusQuery = GetEntityQuery<StatusEffectsComponent>();
         _xformQuery = GetEntityQuery<TransformComponent>();
+        _physicsQuery = GetEntityQuery<PhysicsComponent>();
 
         SubscribeLocalEvent<StationGridAddedEvent>(OnStationGridAdd);
     }
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs:95]: These were deliberately made alphabetical so please keep it alphabetical.

**Comment by Varen** [comment]
Addressed the comments so far. Changed it so you are thrown in a direction away from shuttle center, which is fine for the majority of cases and easier on performance. 
Updated video, it looks nicer this way too.

File: Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs [hunk 3]
```diff
@@ -507,26 +507,31 @@ private void KnockOverKids(TransformComponent xform, ref ValueList<EntityUid> to
...
+        if (tile != null && tile.Value.IsSpace() && _mapManager.TryGetGrid(tile.Value.GridUid, out var grid))
+        {
+            Vector2 direction = -Vector2.UnitY;
+
+            var foo = childXform.LocalPosition - shuttleBody.LocalCenter;
+            _throwing.TryThrow(tossed, foo.Normalized() * 10.0f, 50.0f);
         }
+
+
+
```
Comment (EmoGarbage404) [inline Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs:534]:
```suggestion
```

**Review by EmoGarbage404 — APPROVED** [review]
funny pr

**Pull request merged**